samples/src/App.js

```diff
@@ -32,41 +32,142 @@ function App() {
   const [localMicrophoneTrack, setLocalMicrophoneTrack] = useState();
   const [localDisplayTracks, setLocalDisplayTracks] = useState();
 
+  const [bnrEffect, setBnrEffect] = useState();
+  const [blurEffect, setBlurEffect] = useState();
+  const [vbgEffect, setVirtualBackgroundEffect] = useState();
+  const [videoEffect, setVideoEffect] = useState();
+
+  const [trackEvents, setTrackEvents] = useState([]);
+
+  const updateEvents = (event) => {
+    setTrackEvents(trackEvents => [...trackEvents,{message: event}]);
+  }
+
+  const enableBnr = (enabled) =>{
+    const options = {
+      audioContext,
+      processorUrl,
+      legacyProcessorUrl,
+      mode,
+    };
+    effect = new NoiseReductionEffect(options);
+
+  }
+
+  const enableVirtualBackground = (enabled) => {
+
+    if(!videoEffect) {
+      effect = new VirtualBackgroundEffect({
+        mode: "IMAGE",
+        bgVideoUrl: VIDEO_URL,
+      });
+
+      localCameraTrack.addEffect("virtualBackground", effect)
```

Comment:
should there be a call to `setVideoEffect` in here?

Reply:
i can add it back , previously i remove it based on the review comments from either you or marcin dont remember

Reply:
@bbaldino  my bad we cannot change it directly as addEffect is part of the local-track , I could add it on local-video-track  but not much difference

Reply:
fixed this